Whitespace: Add the `whitespace` rule

Hywan opened the pull request:
Fix #9.
#### Specification

https://github.com/php/php-langspec/blob/master/spec/19-grammar.md#white-space
#### Progression
- [x] White space,
- [x] White space character.

Hywan committed:
Whitespace: Add the `whitespace` rule.

Hywan commented:
@jubianchi Ready for a review!

nikic commented on source/rules/whitespaces.rs:
```diff
@@ -0,0 +1,112 @@
+    for (index, item) in input.iter().enumerate() {
+        if *item != '\n' as u8 &&
+           *item != ' ' as u8 &&
+           *item != '\t' as u8 {
```
What about `\r`?

Hywan replied:
Correct. Missing this is _new-line_ as a rule, not as a token.

nikic commented on source/rules/whitespaces.rs:
```diff
@@ -0,0 +1,99 @@
+
+named!(
+    pub whitespace,
+    re_bytes_find_static!(r"^( |\t|\r\n|\r|\n)+")
```
The extra `\r\n` case here shouldn't be necessary, it's already handled as a combination of `\r` and `\n`. (In this case you likely also don't need a regex)

Hywan committed:
Whitespace: Simplify the `whitespace` rule.

The regular expression is no longer needed because `\r\n` is not a
strict pattern to match since it is covered by `\r` and `\n`
individually.
